pkg/router/routeapihelpers/validation_test.go

```diff
@@ -2316,6 +2479,61 @@ func TestExtendedValidateRoute(t *testing.T) {
 			},
 			expectedErrors: 1,
 		},
+		{
+			// Validate that a key generated from p12 is allowed
+			name: "A keypair containing bag attributes should be cleaned and allowed",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						Certificate: leafCertExtractedFromP12 + caChainExtractedFromP12,
+						Key:         leafKeyExtractedFromP12,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			// Wrong order of certificates should be rejected
+			name: "A certificate field that has the CA chain first should be rejected",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						Certificate: caChainExtractedFromP12 + leafCertExtractedFromP12,
+						Key:         leafKeyExtractedFromP12,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			// Private key containing the CA chain should be rejected
+			name: "A key field containing Public Key/certificate attributes should be rejected",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						Certificate: leafCertExtractedFromP12,
+						Key:         leafKeyExtractedFromP12 + caChainExtractedFromP12,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
```

Comment:
So, this test case is supposed to address the bug in the title, right? The rest of tests cases are here just to confirm the behavior which existed before this PR.

Reply:
> I have generated them as p12 with bag attributes to follow that very specific case as well.

> that creating the tls.crt usingthe wrong order will be rejected (my change doesn't do it, but it is done by X509 validation from Go, so just be sure this case is covered and doesn't regress)

> that the legacy bundled certificate doesn't regress

Thanks for adding more test scenarios!

> that the case from this bug is covered (private key containing certificates)

Right. I asked the question because from reading the bug comments it didn't seem 100% certain that this is the only issue with the customer configuration.

Reply:
No, the PR description LGTM.